Add a new parameter to avoid starting a replacement for lost allocs (#19101)
hashicorp/nomad: website/content/docs/job-specification/group.mdx

  rescheduling strategy. Nomad will then attempt to schedule the task on another
  node if any of the group allocation statuses become "failed".

- `avoid_reschedule_on_lost` `(bool: false)` - Defines the reschedule behaviour 
  of an allocation when the node it is running on becomes disconnect or lost. 
  When enabled, if the node it is running on becomes disconnected 
  or goes down, this allocations wont be rescheduled and will show up as `unknown`
  until the node comes back up or it is manually restarted. 
  This behaviour will only modify the reschedule process on the server.
  To modify the allocation behaviour on the client, see `stop_after_client_disconnect`.

  Setting `max_client_disconnect` and `avoid_reschedule_on_list=true` at the same time 
  requires that [rescheduling is disabled entirely][] If [`max_client_disconnect`](#max_client_disconnect) 
  is set and `avoid_reschedule_on_lost=true`, allocations on disconnected nodes will be
  `unknown` until the `max_client_disconnect` window expires, at which point they 

>   become `lost` but will not be rescheduled. 

Reviewer: Notes from our sidebar conversation:
* We'll need to correct this line.
* And add some documentation on how to recover the unknown allocation if you want (maybe just re-run the job)
* Once those are done this should be LGTM!

  rescheduling strategy. Nomad will then attempt to schedule the task on another
  node if any of the group allocation statuses become "failed".

> - `prevent_reschedule_on_lost` `(bool: false)` - Defines the reschedule behaviour 
>   of an allocation when the node it is running on becomes disconnect or lost. 

Reviewer 2: ```suggestion
- `prevent_reschedule_on_lost` `(bool: false)` - Defines the reschedule behaviour 
  of an allocation when the node it is running on misses heartbeats. 
```

>   When enabled, if the node it is running on becomes disconnected 
>   or goes down, this allocations wont be rescheduled and will show up as `unknown`
>   until the node comes back up or it is manually restarted. 

Reviewer 2: ```suggestion
  When enabled, the allocation is not rescheduled and remains in the `unknown` status
  until the node comes back up or the allocation is manually restarted. 
```

>   This behaviour will only modify the reschedule process on the server.
>   To modify the allocation behaviour on the client, see 

Reviewer 2: ```suggestion
  This behaviour only modifies the reschedule process on the server.
  To modify the allocation behaviour on the client, use 
```

  [`stop_after_client_disconnect`](#stop_after_client_disconnect) .

>   Setting `max_client_disconnect` and `prevent_reschedule_on_lost=true` at the same 

Reviewer 2: ```suggestion
  Setting `max_client_disconnect` and `prevent_reschedule_on_lost = true` at the same 
```

>   time requires that [rescheduling is disabled entirely][].

Reviewer 2: Broken link. Where should `rescheduling is disabled entirely` point to?

>   If [`max_client_disconnect`](#max_client_disconnect) is set and 
>   `prevent_reschedule_on_lost=true`, allocations on disconnected nodes will be
>   `unknown` until the `max_client_disconnect` window expires, at which point 

Reviewer 2: ```suggestion
  If [`max_client_disconnect`](#max_client_disconnect) is set and 
  `prevent_reschedule_on_lost = true`, allocations on disconnected nodes remains with status
  `unknown` until the `max_client_disconnect` window expires, at which point 
```

>   the node will be transition from `disconnected` to `down`. The allocation 
>   will remain as `unknown` and won't be rescheduled.

Reviewer 2: ```suggestion
  the node transitions from `disconnected` to `down`. The allocation 
  remain in the `unknown` status and is not rescheduled.
```

>   will remain as `unknown` and won't be rescheduled.

Reviewer 2: This interaction between `prevent_reschedule_on_lost` and `max_client_disconnect` should probably be documented in both places. So perhaps we should move this part to a separate section and link to there from both configs?

>   To get the job up and running again it has to be manually stoped and run again

Reviewer: ```suggestion
  To get the job up and running again it has to be manually stopped and run again
```

>   To get the job up and running again it has to be manually stopped and run again
>   or [restarted][restart] using the `reschedule` option:
> 
>   ```plaintext
>   `nomad job restart -reschedule <job>`
>   ```

Reviewer 2: ```suggestion
  The `unknown` allocation has to be manually stopped to run it again.

  ```plaintext
  `nomad alloc stop  <alloc ID>`
  ```
```

I don't think we want to restart entire job if just a few allocations disconnect?
